diff --git a/Makefile b/Makefile
index 323a4c3bb..3d55fd079 100644
--- a/Makefile
+++ b/Makefile
@@ -18,7 +18,7 @@ ifeq ($(OS), Windows_NT)
     FPIC=
 else
     # let's assume "normal" UNIX such as linux
-    COMMON_LD_FLAGS=-ldl -lpthread -lz
+    COMMON_LD_FLAGS=-ldl -lpthread -lz -lperfhalide -lpapi
     FPIC=-fPIC
 ifeq ($(UNAME), Darwin)
     SHARED_EXT=dylib
@@ -401,6 +401,7 @@ SOURCE_FILES = \
   ParamMap.cpp \
   Parameter.cpp \
   PartitionLoops.cpp \
+  PerfHalideProfiling.cpp \
   Pipeline.cpp \
   Prefetch.cpp \
   PrintLoopNest.cpp \
@@ -549,6 +550,7 @@ HEADER_FILES = \
   ParamMap.h \
   Parameter.h \
   PartitionLoops.h \
+  PerfHalideProfiling.h \
   Pipeline.h \
   Prefetch.h \
   Profiling.h \
@@ -652,6 +654,7 @@ RUNTIME_CPP_COMPONENTS = \
   posix_tempfile \
   posix_threads \
   powerpc_cpu_features \
+  perf_halide \
   prefetch \
   profiler \
   profiler_inlined \
