diff --git a/src/CodeGen_Internal.cpp b/src/CodeGen_Internal.cpp
index 5d53953..1d4c84d 100644
--- a/src/CodeGen_Internal.cpp
+++ b/src/CodeGen_Internal.cpp
@@ -160,6 +160,8 @@ bool function_takes_user_context(const std::string &name) {
         "halide_free",
         "halide_malloc",
         "halide_print",
+        "halide_papi_pipeline_start",
+        "halide_papi_pipeline_stop",
         "halide_profiler_memory_allocate",
         "halide_profiler_memory_free",
         "halide_profiler_pipeline_start",
