diff --git a/src/Func.cpp b/src/Func.cpp
index 0831afde7..84c3f2795 100644
--- a/src/Func.cpp
+++ b/src/Func.cpp
@@ -2413,6 +2413,18 @@ Func &Func::trace_realizations() {
     return *this;
 }
 
+Func &Func::profile(int level, bool show_threads, bool enable) {
+    invalidate_cache();
+    func.profile(level, show_threads, enable);
+    return *this;
+}
+
+Func &Func::profile_at(Internal::Function &parent, int level, bool show_threads, bool enable) {
+    invalidate_cache();
+    func.profile_at(parent, level, show_threads, enable);
+    return *this;
+}
+
 void Func::debug_to_file(const string &filename) {
     invalidate_cache();
     func.debug_file() = filename;
