diff --git a/src/Func.h b/src/Func.h
index 37b4a1d53..2e3d3f607 100644
--- a/src/Func.h
+++ b/src/Func.h
@@ -2243,6 +2243,12 @@ public:
      * halide_trace. */
     Func &trace_realizations();
 
+    /** Profile this function. */
+    Func &profile(int level = 1, bool show_threads = false, bool enable = true);
+
+    /** Profile this function inside some parent. */
+    Func &profile_in(Internal::Function &parent, int level = 1, bool show_threads = false, bool enable = true);
+
     /** Get a handle on the internal halide function that this Func
      * represents. Useful if you want to do introspection on Halide
      * functions */
