diff --git a/src/Function.cpp b/src/Function.cpp
index 1a2569b9a..52af9b787 100644
--- a/src/Function.cpp
+++ b/src/Function.cpp
@@ -13,6 +13,7 @@
 #include "IREquality.h"
 #include "IROperator.h"
 #include "IRPrinter.h"
+#include "PAPIProfiling.h"
 #include "ParallelRVar.h"
 #include "Var.h"
 
@@ -29,6 +30,8 @@ typedef map<FunctionPtr, FunctionPtr> DeepCopyMap;
 
 struct FunctionContents;
 
+extern std::vector<std::tuple<std::string, int, bool, std::string>> papi_profiler_defs;
+
 namespace {
 // Weaken all the references to a particular Function to break
 // reference cycles. Also count the number of references found.
@@ -885,6 +888,22 @@ void Function::trace_stores() {
 void Function::trace_realizations() {
     contents->trace_realizations = true;
 }
+
+void Function::profile(int level, bool show_threads, bool enable) {
+    if(show_threads) {
+        level |= PROFILE_SHOW_THREADS;
+    }
+
+    papi_profiler_defs.push_back(std::make_tuple(contents->name, level, enable, ""));
+}
+void Function::profile_in(Function &parent, int level, bool show_threads, bool enable) {
+    if(show_threads) {
+        level |= PROFILE_SHOW_THREADS;
+    }
+
+    papi_profiler_defs.push_back(std::make_tuple(contents->name, level, enable, parent.name()));
+}
+
 bool Function::is_tracing_loads() const {
     return contents->trace_loads;
 }
