diff --git a/src/Func.cpp b/src/Func.cpp
index d582f7818..211a24cfe 100644
--- a/src/Func.cpp
+++ b/src/Func.cpp
@@ -2470,6 +2470,18 @@ Func &Func::trace_realizations() {
     return *this;
 }
 
+Func &Func::profile(int level, bool show_threads, bool enable) {
+    invalidate_cache();
+    func.profile(level, show_threads, enable);
+    return *this;
+}
+
+Func &Func::profile_at(Internal::Function &parent, int level, bool show_threads, bool enable) {
+    invalidate_cache();
+    func.profile_at(parent, level, show_threads, enable);
+    return *this;
+}
+
 Func &Func::add_trace_tag(const std::string &trace_tag) {
     invalidate_cache();
     func.add_trace_tag(trace_tag);
