diff --git a/src/Function.cpp b/src/Function.cpp
index d9e57c2c2..64f64446c 100644
--- a/src/Function.cpp
+++ b/src/Function.cpp
@@ -92,6 +92,8 @@ struct FunctionContents {
 
     bool frozen = false;
 
+    bool profile = false;
+
     void accept(IRVisitor *visitor) const {
         func_schedule.accept(visitor);
 
@@ -345,6 +347,7 @@ void Function::deep_copy(FunctionPtr copy, DeepCopyMap &copied_map) const {
     copy->frozen = contents->frozen;
     copy->output_buffers = contents->output_buffers;
     copy->func_schedule = contents->func_schedule.deep_copy(copied_map);
+    copy->profile = contents->profile;
 
     // Copy the pure definition
     if (contents->init_def.defined()) {
@@ -887,6 +890,9 @@ void Function::trace_stores() {
 void Function::trace_realizations() {
     contents->trace_realizations = true;
 }
+void Function::profile() {
+    contents->profile = true;
+}
 void Function::add_trace_tag(const std::string &trace_tag) {
     contents->trace_tags.push_back(trace_tag);
 }
@@ -900,6 +906,9 @@ bool Function::is_tracing_stores() const {
 bool Function::is_tracing_realizations() const {
     return contents->trace_realizations;
 }
+bool Function::is_profiling() const {
+    return contents->profile;
+}
 const std::vector<std::string> &Function::get_trace_tags() const {
     return contents->trace_tags;
 }
