diff --git a/src/Func.h b/src/Func.h
index c3a62c559..36b92f9c6 100644
--- a/src/Func.h
+++ b/src/Func.h
@@ -2292,6 +2292,9 @@ public:
      * halide_trace. */
     Func &trace_realizations();
 
+    /** Profile this function. */
+    Func &profile(int level);
+
     /** Add a string of arbitrary text that will be passed thru to trace
      * inspection code if the Func is realized in trace mode. (Funcs that are
      * inlined won't have their tags emitted.) Ignored entirely if
