diff --git a/src/IR.h b/src/IR.h
index 1cf3bd2e0..8a76b2dec 100644
--- a/src/IR.h
+++ b/src/IR.h
@@ -291,11 +291,12 @@ struct ProducerConsumer : public StmtNode<ProducerConsumer> {
     std::string name;
     bool is_producer;
     Stmt body;
+    bool must_profile;
 
-    static Stmt make(const std::string &name, bool is_producer, Stmt body);
+    static Stmt make(const std::string &name, bool is_producer, Stmt body, bool must_profile);
 
-    static Stmt make_produce(const std::string &name, Stmt body);
-    static Stmt make_consume(const std::string &name, Stmt body);
+    static Stmt make_produce(const std::string &name, Stmt body, bool must_profile);
+    static Stmt make_consume(const std::string &name, Stmt body, bool must_profile);
 
     static const IRNodeType _node_type = IRNodeType::ProducerConsumer;
 };
