diff --git a/src/IRPrinter.cpp b/src/IRPrinter.cpp
index 57c585d46..16c2969cf 100644
--- a/src/IRPrinter.cpp
+++ b/src/IRPrinter.cpp
@@ -156,8 +156,8 @@ void IRPrinter::test() {
     Stmt store2 = Store::make("out", call + 1, x, Parameter(), const_true());
     Stmt for_loop2 = For::make("x", 0, y, ForType::Vectorized , DeviceAPI::Host, store2);
 
-    Stmt producer = ProducerConsumer::make_produce("buf", for_loop);
-    Stmt consumer = ProducerConsumer::make_consume("buf", for_loop2);
+    Stmt producer = ProducerConsumer::make_produce("buf", for_loop, false);
+    Stmt consumer = ProducerConsumer::make_consume("buf", for_loop2, false);
     Stmt pipeline = Block::make(producer, consumer);
 
     Stmt assertion = AssertStmt::make(y >= 3, Call::make(Int(32), "halide_error_param_too_small_i64",
