diff --git a/src/LoopCarry.cpp b/src/LoopCarry.cpp
index 8e59aa4da..29faa9cf6 100644
--- a/src/LoopCarry.cpp
+++ b/src/LoopCarry.cpp
@@ -512,7 +512,7 @@ class LoopCarry : public IRMutator2 {
         } else {
             ScopedBinding<> bind(in_consume, op->name);
             Stmt body = mutate(op->body);
-            return ProducerConsumer::make(op->name, op->is_producer, body);
+            return ProducerConsumer::make(op->name, op->is_producer, body, op->must_profile);
         }
     }
 
