diff --git a/src/Lower.cpp b/src/Lower.cpp
index 906115f78..87d8b666e 100644
--- a/src/Lower.cpp
+++ b/src/Lower.cpp
@@ -35,6 +35,7 @@
 #include "LoopCarry.h"
 #include "LowerWarpShuffles.h"
 #include "Memoization.h"
+#include "PAPIProfiling.h"
 #include "PartitionLoops.h"
 #include "Prefetch.h"
 #include "Profiling.h"
@@ -290,6 +291,12 @@ Module lower(const vector<Function> &output_funcs, const string &pipeline_name,
         debug(2) << "Lowering after injecting profiling:\n" << s << "\n\n";
     }
 
+    if (t.has_feature(Target::PAPI)) {
+        debug(1) << "Injecting PAPI profiling...\n";
+        s = inject_papi_profiling(s, pipeline_name);
+        debug(2) << "Lowering after injecting PAPI profiling:\n" << s << "\n\n";
+    }
+
     if (t.has_feature(Target::FuzzFloatStores)) {
         debug(1) << "Fuzzing floating point stores...\n";
         s = fuzz_float_stores(s);
