diff --git a/src/Lower.cpp b/src/Lower.cpp
index a25d18de3..1f6fc5bff 100644
--- a/src/Lower.cpp
+++ b/src/Lower.cpp
@@ -40,6 +40,7 @@
 #include "LoopCarry.h"
 #include "LowerWarpShuffles.h"
 #include "Memoization.h"
+#include "PAPIProfiling.h"
 #include "PartitionLoops.h"
 #include "Prefetch.h"
 #include "Profiling.h"
@@ -406,6 +407,12 @@ Module lower(const vector<Function> &output_funcs,
                  << s << "\n\n";
     }
 
+    if (t.has_feature(Target::PAPI)) {
+        debug(1) << "Injecting PAPI profiling...\n";
+        s = inject_papi_profiling(s, pipeline_name);
+        debug(2) << "Lowering after injecting PAPI profiling:\n" << s << "\n\n";
+    }
+
     if (t.has_feature(Target::CUDA)) {
         debug(1) << "Injecting warp shuffles...\n";
         s = lower_warp_shuffles(s);
