diff --git a/src/Lower.cpp b/src/Lower.cpp
index bed3fe4b4..ed0d5ea35 100644
--- a/src/Lower.cpp
+++ b/src/Lower.cpp
@@ -36,6 +36,7 @@
 #include "LowerWarpShuffles.h"
 #include "Memoization.h"
 #include "PartitionLoops.h"
+#include "PerfHalideProfiling.h"
 #include "Prefetch.h"
 #include "Profiling.h"
 #include "Qualify.h"
@@ -294,6 +295,12 @@ Module lower(const vector<Function> &output_funcs, const string &pipeline_name,
         debug(2) << "Lowering after injecting profiling:\n" << s << "\n\n";
     }
 
+    if (t.has_feature(Target::PerfHalideProfile)) {
+        debug(1) << "Injecting perf_halide profiling...\n";
+        s = inject_perf_halide_profiling(s, pipeline_name);
+        debug(2) << "Lowering after injecting perf_halide profiling:\n" << s << "\n\n";
+    }
+
     if (t.has_feature(Target::FuzzFloatStores)) {
         debug(1) << "Fuzzing floating point stores...\n";
         s = fuzz_float_stores(s);
