diff --git a/src/RemoveUndef.cpp b/src/RemoveUndef.cpp
index 21b695b05..20c5bd0dd 100644
--- a/src/RemoveUndef.cpp
+++ b/src/RemoveUndef.cpp
@@ -240,7 +240,7 @@ private:
         if (body.same_as(op->body)) {
             return op;
         } else {
-            return ProducerConsumer::make(op->name, op->is_producer, body);
+            return ProducerConsumer::make(op->name, op->is_producer, body, op->must_profile);
         }
     }
 
