diff --git a/src/ScheduleFunctions.cpp b/src/ScheduleFunctions.cpp
index 205881904..da34e48dc 100644
--- a/src/ScheduleFunctions.cpp
+++ b/src/ScheduleFunctions.cpp
@@ -807,11 +807,11 @@ private:
             internal_assert(realization.second.defined());
             producer = realization.second;
         }
-        producer = ProducerConsumer::make_produce(func.name(), producer);
+        producer = ProducerConsumer::make_produce(func.name(), producer, func.is_profiling());
 
         // Outputs don't have consume nodes
         if (!is_output) {
-            consumer = ProducerConsumer::make_consume(func.name(), consumer);
+            consumer = ProducerConsumer::make_consume(func.name(), consumer, func.is_profiling());
         }
 
         if (is_no_op(consumer)) {
@@ -1195,7 +1195,7 @@ private:
         Stmt consume = s;
         for (size_t i = 0; i < group.size(); ++i) {
             if (!skip[group[i].name()]) {
-                consume = ProducerConsumer::make(group[i].name(), false, consume);
+                consume = ProducerConsumer::make(group[i].name(), false, consume, group[i].is_profiling());
             }
         }
 
@@ -1254,7 +1254,7 @@ private:
         // Add the producer nodes.
         for (size_t i = 0; i < group.size(); ++i) {
             if (!skip[group[i].name()]) {
-                produce = ProducerConsumer::make(group[i].name(), true, produce);
+                produce = ProducerConsumer::make(group[i].name(), true, produce, group[i].is_profiling());
             }
         }
 
