diff --git a/src/Simplify_Stmts.cpp b/src/Simplify_Stmts.cpp
index 152abfd2f..701113715 100644
--- a/src/Simplify_Stmts.cpp
+++ b/src/Simplify_Stmts.cpp
@@ -63,7 +63,7 @@ Stmt Simplify::visit(const IfThenElse *op) {
                then_pc->name == else_pc->name &&
                then_pc->is_producer == else_pc->is_producer) {
         return ProducerConsumer::make(then_pc->name, then_pc->is_producer,
-                                      mutate(IfThenElse::make(condition, then_pc->body, else_pc->body)));
+                                      mutate(IfThenElse::make(condition, then_pc->body, else_pc->body)), then_pc->must_profile);
     } else if (then_block &&
                else_block &&
                equal(then_block->first, else_block->first)) {
@@ -267,7 +267,7 @@ Stmt Simplify::visit(const ProducerConsumer *op) {
     } else if (body.same_as(op->body)) {
         return op;
     } else {
-        return ProducerConsumer::make(op->name, op->is_producer, body);
+        return ProducerConsumer::make(op->name, op->is_producer, body, op->must_profile);
     }
 }
 
