diff --git a/src/Target.cpp b/src/Target.cpp
index 64ea2882c..02ee558d3 100644
--- a/src/Target.cpp
+++ b/src/Target.cpp
@@ -266,6 +266,7 @@ const std::map<std::string, Target::Feature> feature_name_map = {
     {"trace_realizations", Target::TraceRealizations},
     {"strict_float", Target::StrictFloat},
     {"legacy_buffer_wrappers", Target::LegacyBufferWrappers},
+    {"perf_halide", Target::PerfHalideProfile},
 };
 
 bool lookup_feature(const std::string &tok, Target::Feature &result) {
