diff --git a/src/Target.h b/src/Target.h
index 313c106b8..e3ca92bcf 100644
--- a/src/Target.h
+++ b/src/Target.h
@@ -71,6 +71,7 @@ struct Target {
         OpenGLCompute = halide_target_feature_openglcompute,
         UserContext = halide_target_feature_user_context,
         Matlab = halide_target_feature_matlab,
+        PAPI = halide_target_feature_papi,
         Profile = halide_target_feature_profile,
         NoRuntime = halide_target_feature_no_runtime,
         Metal = halide_target_feature_metal,
