diff --git a/src/Target.h b/src/Target.h
index cde96216b..9bbb6fb6d 100644
--- a/src/Target.h
+++ b/src/Target.h
@@ -87,6 +87,7 @@ struct Target {
         EGL = halide_target_feature_egl,
         UserContext = halide_target_feature_user_context,
         Matlab = halide_target_feature_matlab,
+        PAPI = halide_target_feature_papi,
         Profile = halide_target_feature_profile,
         NoRuntime = halide_target_feature_no_runtime,
         Metal = halide_target_feature_metal,
