diff --git a/src/Target.h b/src/Target.h
index 8cbfba903..6759fb46e 100644
--- a/src/Target.h
+++ b/src/Target.h
@@ -64,6 +64,7 @@ struct Target {
         OpenGLCompute = halide_target_feature_openglcompute,
         UserContext = halide_target_feature_user_context,
         Matlab = halide_target_feature_matlab,
+        PerfHalideProfile = halide_target_feature_perf_halide,
         Profile = halide_target_feature_profile,
         NoRuntime = halide_target_feature_no_runtime,
         Metal = halide_target_feature_metal,
