diff --git a/src/Target.cpp b/src/Target.cpp
index b61612894..4fa7406bc 100644
--- a/src/Target.cpp
+++ b/src/Target.cpp
@@ -242,6 +242,7 @@ const std::map<std::string, Target::Feature> feature_name_map = {
     {"openglcompute", Target::OpenGLCompute},
     {"user_context", Target::UserContext},
     {"matlab", Target::Matlab},
+    {"papi", Target::PAPI},
     {"profile", Target::Profile},
     {"no_runtime", Target::NoRuntime},
     {"metal", Target::Metal},
