diff --git a/src/Target.h b/src/Target.h
index 37e5ae3fd..4b657c2e3 100644
--- a/src/Target.h
+++ b/src/Target.h
@@ -64,6 +64,7 @@ struct Target {
         OpenGLCompute = halide_target_feature_openglcompute,
         UserContext = halide_target_feature_user_context,
         Matlab = halide_target_feature_matlab,
+        PAPI = halide_target_feature_papi,
         Profile = halide_target_feature_profile,
         NoRuntime = halide_target_feature_no_runtime,
         Metal = halide_target_feature_metal,
