diff --git a/src/Target.cpp b/src/Target.cpp
index 21e0cac67..d49a57fd9 100644
--- a/src/Target.cpp
+++ b/src/Target.cpp
@@ -272,6 +272,7 @@ const std::map<std::string, Target::Feature> feature_name_map = {
     {"openglcompute", Target::OpenGLCompute},
     {"user_context", Target::UserContext},
     {"matlab", Target::Matlab},
+    {"papi", Target::PAPI},
     {"profile", Target::Profile},
     {"no_runtime", Target::NoRuntime},
     {"metal", Target::Metal},
