diff --git a/src/runtime/runtime_internal.h b/src/runtime/runtime_internal.h
index 571351a3a..7296e0fe9 100644
--- a/src/runtime/runtime_internal.h
+++ b/src/runtime/runtime_internal.h
@@ -165,6 +165,21 @@ WEAK int halide_trace_helper(void *user_context,
                              int code,
                              int parent_id, int value_index, int dimensions);
 
+WEAK int halide_papi_pipeline_start(void *user_context,
+                                    const char *pipeline_name,
+                                    int num_funcs,
+                                    const uint64_t *func_names,
+                                    const int64_t *func_parents_prod,
+                                    const int64_t *func_parents_cons,
+                                    const uint64_t *func_show_threads_prod,
+                                    const uint64_t *func_show_threads_cons);
+WEAK void halide_papi_stack_peak_update(void *user_context, void *pipeline_state, uint64_t *f_values);
+WEAK void halide_papi_memory_allocate(void *user_context, void *pipeline_state, int func_id, uint64_t incr);
+WEAK void halide_papi_memory_free(void *user_context, void *pipeline_state, int func_id, uint64_t decr);
+WEAK void halide_papi_report(void *user_context);
+WEAK void halide_papi_reset();
+WEAK void halide_papi_pipeline_end(void *user_context, void *state);
+
 }  // extern "C"
 
 /** A macro that calls halide_print if the supplied condition is
