diff --git a/src/runtime/runtime_api.cpp b/src/runtime/runtime_api.cpp
index 5c9b9a6c0..347ea1e90 100644
--- a/src/runtime/runtime_api.cpp
+++ b/src/runtime/runtime_api.cpp
@@ -150,6 +150,11 @@ extern "C" __attribute__((used)) void *halide_runtime_api_functions[] = {
     (void *)&halide_openglcompute_device_interface,
     (void *)&halide_openglcompute_initialize_kernels,
     (void *)&halide_openglcompute_run,
+    (void *)&halide_papi_get_pipeline_state,
+    (void *)&halide_papi_get_state,
+    (void *)&halide_papi_pipeline_start,
+    (void *)&halide_papi_report,
+    (void *)&halide_papi_reset,
     (void *)&halide_pointer_to_string,
     (void *)&halide_print,
     (void *)&halide_profiler_get_pipeline_state,
