diff --git a/src/runtime/runtime_internal.h b/src/runtime/runtime_internal.h
index 7d818d594..96704f54e 100644
--- a/src/runtime/runtime_internal.h
+++ b/src/runtime/runtime_internal.h
@@ -155,6 +155,14 @@ WEAK int halide_trace_helper(void *user_context,
                              int parent_id, int value_index, int dimensions,
                              const char *trace_tag);
 
+WEAK int halide_papi_pipeline_start(void *user_context, const char *pipeline_name, int num_funcs, const uint64_t *func_names);
+WEAK void halide_papi_stack_peak_update(void *user_context, void *pipeline_state, uint64_t *f_values);
+WEAK void halide_papi_memory_allocate(void *user_context, void *pipeline_state, int func_id, uint64_t incr);
+WEAK void halide_papi_memory_free(void *user_context, void *pipeline_state, int func_id, uint64_t decr);
+WEAK void halide_papi_report(void *user_context);
+WEAK void halide_papi_reset();
+WEAK void halide_papi_pipeline_end(void *user_context, void *state);
+
 }  // extern "C"
 
 /** A macro that calls halide_print if the supplied condition is
