diff --git a/src/runtime/runtime_internal.h b/src/runtime/runtime_internal.h
index a0fa5aead..c7fb46bbe 100644
--- a/src/runtime/runtime_internal.h
+++ b/src/runtime/runtime_internal.h
@@ -134,6 +134,16 @@ WEAK int halide_profiler_pipeline_start(void *user_context,
                                         const char *pipeline_name,
                                         int num_funcs,
                                         const uint64_t *func_names);
+
+WEAK int perf_halide_pipeline_start(void *user_context, const char *pipeline_name, int num_funcs, const uint64_t *func_names);
+WEAK void perf_halide_stack_peak_update(void *user_context, void *pipeline_state, uint64_t *f_values);
+WEAK void perf_halide_memory_allocate(void *user_context, void *pipeline_state, int func_id, uint64_t incr);
+WEAK void perf_halide_memory_free(void *user_context, void *pipeline_state, int func_id, uint64_t decr);
+WEAK void perf_halide_report(void *user_context);
+WEAK void perf_halide_reset();
+WEAK void perf_halide_shutdown();
+WEAK void perf_halide_pipeline_end(void *user_context, void *state);
+
 WEAK int halide_host_cpu_count();
 
 WEAK int halide_device_and_host_malloc(void *user_context, struct halide_buffer_t *buf,
